lib/ui/viewer/search/search_widget.dart

```diff
@@ -139,9 +143,36 @@ class _SearchWidgetState extends State<SearchWidget> {
     );
   }
 
+  @override
+  void dispose() {
+    _debouncer.cancel();
+    super.dispose();
+  }
+
   Future<List<SearchResult>> getSearchResultsForQuery(String query) async {
     final List<SearchResult> allResults = [];
+    if (query.isEmpty) {
+      if (_debouncer.debounceTimer != null &&
```

Comment:
You could extract this logic to a function inside Debouncer? So that it'd simply be `debouncer.isActive()`